pkg/operator/resources/batchapi/api.go

```diff
@@ -77,11 +77,19 @@ func UpdateAPI(apiConfig *userconfig.API, projectID string) (*spec.API, string,
 
 		err = applyK8sResources(api, prevVirtualService)
 		if err != nil {
-			return nil, "", err
+			rollBackErr := rollBack(prevVirtualService)
+			if rollBackErr != nil {
+				return nil, "", err
```

Comment:
I think it might be good to return both errors, and say something along the lines of “update failed” with the error, and “During rollback, another error occurred” with that error